packages/core/src/readers/LlamaParseReader.ts

```diff
@@ -350,18 +350,19 @@ export class LlamaParseReader extends FileReader {
     return images;
   }
 
-  private async getMimeType(data: Buffer): Promise<string> {
-    const mimes = filetypemime(data);
-    const validMime = mimes.find((mime) =>
-      Object.values(SupportedFiles).includes(mime),
-    );
-    if (!validMime) {
+  private async getMimeType(
+    data: Buffer,
+  ): Promise<{ mimeType: string; extension: string }> {
+    const mimes = filetypemime(data); // Get an array of possible MIME types
```

Comment:
@KindOfAScam don't you skip now the other valid mime types?

Reply:
Yes but on purpose.
 
Before it was iterating through the list of potential mime types and picking the first one that matches a supported extension. This could result in an unsupported mime type being assigned just because a less likely mime type is supported.

Now it's choosing the most likely mime type and checks if it fits a  supported extension. That's also equivalent to python version.